### PR TITLE
feature: make envs replaceable with envsubt for search and chat builds

### PR DESCRIPTION
**Pull request by skeptrunedev**
## The Point

In a deployment, you can pull down the bundled js for `search` and `chat` then replace the static env values with `set -a && source .env && set +a` and `envsubst < index.html.template > dist/index.html && rm index.html.template` for each bundle respectively. 

I used `vite-plugin-runtime-env`  (tho poorly named) from [this repo](https://github.com/micha149/vite-plugin-runtime-env/blob/master/src/index.ts) to do this. I think it's a solid implementation. 

### Chat

```
cat .env.chat > ./chat/.env
cd chat
yarn build
cp dist/index.html index.html.template
set -a && source .env && set +a
envsubst < index.html.template > dist/index.html && rm index.html.template
 ```

### Search

```
cat .env.search > ./search/.env
cd search
yarn build
cp dist/index.html index.html.template
set -a && source .env && set +a
envsubst < index.html.template > dist/index.html && rm index.html.template
```

### CODE REVIEW

File: .env.chat
[the remark this answers is not in the document]
Comment (skeptrunedev) [reply .env.chat:1]: At the top of a citation chunk, there's "front matter" which is like the keys from the chunk object that get shown and these set those.

[the remark this answers is not in the document]
Comment (skeptrunedev) [reply .env.chat:1]: I can attempt to explain in the getting started guide

**Comment by skeptrunedev** [comment]
@tfolbrecht in the build, it looks like this: 

image


And then, in the `index.html`: 

image

**Review by tfolbrecht — APPROVED** [review]
Clean!

**Pull request merged**